### PR TITLE
Call attach-router if interface ip does not exist

### LINKED ISSUE #1249: Weave expose fails randomly on node reboot
Weave bridge creation is sometimes really slow and this causes expose to fail. 

Related: https://github.com/weaveworks/weave/issues/2479

### PR DESCRIPTION
**Pull request by jakolehm**
Fixes #1249

### CODE REVIEW

**Comment by jakolehm** [comment]
@jnummelin @SpComb ping

File: agent/lib/kontena/network_adapters/weave.rb [hunk 1]
```diff
@@ -201,6 +201,7 @@ def start(info)
         end
       end
 
+      attach_router unless interface_ip('weave')
```
Comment (SpComb) [inline agent/lib/kontena/network_adapters/weave.rb]: What happens if the weave interface exists, but it doesn't have any address configured?

Looking at  #1165 and `Errno::EADDRNOTAVAIL`...

Comment (SpComb) [inline agent/lib/kontena/network_adapters/weave.rb]: Like this:

```
$ ip addr show weave
26: weave: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1410 qdisc noqueue state UP group default qlen 1000
    link/ether c6:3f:e0:48:25:8d brd ff:ff:ff:ff:ff:ff
    inet6 fe80::c43f:e0ff:fe48:258d/64 scope link 
       valid_lft forever preferred_lft forever
```

Giving:

```
kontena-agent | E, [2016-10-31T07:29:52.586188 #1] ERROR -- Kontena::NetworkAdapters::Weave: Errno::EADDRNOTAVAIL: Address not available
kontena-agent | D, [2016-10-31T07:29:52.586289 #1] DEBUG -- NetworkAdapters::Weave: /app/lib/kontena/helpers/iface_helper.rb:21:in `ioctl'
kontena-agent | /app/lib/kontena/helpers/iface_helper.rb:21:in `interface_ip'
kontena-agent | /app/lib/kontena/helpers/iface_helper.rb:32:in `interface_ip'
kontena-agent | /app/lib/kontena/network_adapters/weave.rb:204:in `start'
kontena-agent | /usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `public_send'
kontena-agent | /usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `dispatch'
kontena-agent | /usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/call/async.rb:7:in `dispatch'
kontena-agent | /usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:50:in `block in dispatch'
kontena-agent | /usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:76:in `block in task'
kontena-agent | /usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:339:in `block in task'
kontena-agent | /usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:44:in `block in initialize'
kontena-agent | /usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/task/fibered.rb:14:in `block in create'
```

That doesn't happen normally, but if something fails between the `launch-router`/`attach-router` creating the bridge, and then the `expose` configuring it.

Comment (jnummelin) [inline agent/lib/kontena/network_adapters/weave.rb]: If weave interface exist without address and agent heals it (with a rescue from `Errno::EADDRNOTAVAIL`) for some reason the "healed" weave interface seems to be slightly different compared when it's completely re-created:
```
core@quiet-frost-93 ~ $ ip addr show dev weave
8: weave: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1410 qdisc noqueue state UP group default qlen 1000
    link/ether be:f4:ed:2d:50:04 brd ff:ff:ff:ff:ff:ff
    inet 10.81.0.1/19 scope global weave
       valid_lft forever preferred_lft forever
```
vs. normally created interface:
```
core@quiet-frost-93 ~ $ ip addr show dev weave
8: weave: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1410 qdisc noqueue state UP group default qlen 1000
    link/ether be:f4:ed:2d:50:04 brd ff:ff:ff:ff:ff:ff
    inet 10.81.0.1/19 scope global weave
       valid_lft forever preferred_lft forever
    inet6 fe80::bcf4:edff:fe2d:5004/64 scope link 
       valid_lft forever preferred_lft forever
```

So it misses IPv6 config. As weave (and Kontena) doesn't support IPv6 this should not be an issue.

Comment (jakolehm) [inline agent/lib/kontena/network_adapters/weave.rb]: All agree that this is ok / improvement over current buggy behaviour? :)

Comment (jnummelin) [inline agent/lib/kontena/network_adapters/weave.rb]: Yes. I tried 20+ reboots with this PR and in all of those everything worked as expected.

Comment (SpComb) [inline agent/lib/kontena/network_adapters/weave.rb]: I suggest fixing the `interface_ip()` `rescue Errno::EADDRNOTAVAIL` `return nil` behavior and then LGTM

Comment (jakolehm) [inline agent/lib/kontena/network_adapters/weave.rb]: Isn't that a separate issue?

Comment (SpComb) [inline agent/lib/kontena/network_adapters/weave.rb]: Submitted a PR as #1256

**Pull request merged**